packages/medusa/src/services/product.js

```diff
@@ -388,7 +388,9 @@ class ProductService extends BaseService {
    * @return {Promise} an array of variants
    */
   async retrieveVariants(productId) {
-    const product = await this.retrieve(productId, { relations: ["variants"] })
+    const product = await this.retrieve(productId, {
+      relations: ["variants", "variants.prices"],
+    })
```

Comment:
issue: Add config object containing relations, select, etc. similar to normal retrieve methods